cvmfs/dns.cc

```diff
@@ -1153,9 +1153,9 @@ void HostfileResolver::ParseHostFile() {
     if (hash_pos != string::npos) line = line.substr(0, hash_pos);
 
     // First token is an IP address
-    int ip_start_pos, ip_end_pos;
-    sscanf(line.c_str(), " %n%*s%n", &ip_start_pos, &ip_end_pos);
-    if (ip_start_pos == ip_end_pos)
+    int ip_start_pos, ip_end_pos, scan_result;
```

Comment:
These three variables should be initialized. The `sscanf` function can fail after parsing only a subset of the variables.

Reply:
I can initialize and do further checking to be sure, but the flags in sscanf work in a way that either all or none of the variables are going to be filled.